docs/design/plugin_api.md

```diff
@@ -0,0 +1,363 @@
+# Plugin API
+
+## Purpose
+
+This document describes the API used by variant provider plugins.
+The plugins are a central point of the variant specification, defining
+the valid metadata, and providing routines necessary to install
+and build variants.
+
+The plugin API is defined using Python [Protocols](
+https://typing.python.org/en/latest/spec/protocol.html), as well
+as textual description.
+
+
+## Basic design
+
+Every provider plugin operates within a single namespace. The namespace
+is used as key for all plugin-related operations. All the properties
+defined by the plugin are in the plugin's namespace, and the plugin
+defines all the valid feature names and values within that namespace.
+
+Within a single package, only one plugin can be used for a given
+namespace. Attempting to load a second plugin sharing the same namespace
+must cause a fatal error. However, it is possible for multiple plugins
+using the namespace to exist, which implies that they become mutually
+exclusive. For example, this could happen if a plugin becomes
+unmaintained and needs to be forked into a new package.
+
+Plugins are expected to be automatically installed into an isolated
+environment when installing packages, in order to facilitate variant
```

Comment:
I'd leave out "into an isolated environment" here, because it makes it seem mandatory. Same as for the index configuration comment thread below: I'd leave this to the package manager. It can be an isolated install, a non-isolated install, or built into the package manager already for example.

Reply:
Done and done.